src/logics/LALogic.h

```diff
@@ -126,29 +126,37 @@ class LALogic: public Logic
     virtual PTRef getTerm_NumZero() const = 0;
     virtual PTRef getTerm_NumOne() const = 0;
     virtual PTRef getTerm_NumMinusOne() const = 0;
-    virtual PTRef mkNumNeg(PTRef, char **);
-    virtual PTRef mkNumNeg(PTRef tr);
-    virtual PTRef mkNumMinus(const vec<PTRef> &, char **);
-    virtual PTRef mkNumMinus(const vec<PTRef> &args);
-    virtual PTRef mkNumMinus(const PTRef a1, const PTRef a2);
-    virtual PTRef mkNumPlus(const vec<PTRef> &, char **);
-    virtual PTRef mkNumPlus(const vec<PTRef> &args);
-    virtual PTRef mkNumPlus(const std::vector<PTRef> &args);
-    virtual PTRef mkNumPlus(const PTRef p1, const PTRef p2);
-    virtual PTRef mkNumTimes(const vec<PTRef> &, char **);
-    virtual PTRef mkNumTimes(const vec<PTRef> &args);
-    virtual PTRef mkNumTimes(const PTRef p1, const PTRef p2);
-    virtual PTRef mkNumTimes(const std::vector<PTRef> &args);
-    virtual PTRef mkNumDiv(const vec<PTRef> &args) = 0;
+
+    PTRef mkNumNeg(PTRef tr);
+
+    PTRef mkNumMinus(vec<PTRef> && args);
+    PTRef mkNumMinus(vec<PTRef> const & args) { vec<PTRef> tmp; args.copyTo(tmp); return mkNumMinus(std::move(tmp)); }
```

Comment:
Could this be `vec<PTRef const> const & args` ?  I'm not sure if the current implementation of `vec` allows consts as the template parameters, but I think this could be fixed / addressed.

Reply:
As discussed separately, let's keep it as it is.